src/main/java/core/basesyntax/Engine.java

```diff
@@ -1,5 +1,107 @@
 package core.basesyntax;
 
-public class Engine {
+public class Engine implements Cloneable {
 
+    private String typeOfEngine;
+    private int countOfCylinders;
+    private int countOfValves;
+    private double compressionRatio;
+    private String typeOfFuel;
+
+    public Engine(String typeOfEngine, int countOfCylinders,
+                  int countOfValves, double compressionRatio, String typeOfFuel) {
+        this.typeOfEngine = typeOfEngine;
+        this.countOfCylinders = countOfCylinders;
+        this.countOfValves = countOfValves;
+        this.compressionRatio = compressionRatio;
+        this.typeOfFuel = typeOfFuel;
+    }
+
+    public String getTypeOfEngine() {
+        return typeOfEngine;
+    }
+
+    public void setTypeOfEngine(String typeOfEngine) {
+        this.typeOfEngine = typeOfEngine;
+    }
+
+    public int getCountOfCylinders() {
+        return countOfCylinders;
+    }
+
+    public void setCountOfCylinders(int countOfCylinders) {
+        this.countOfCylinders = countOfCylinders;
+    }
+
+    public int getCountOfValves() {
+        return countOfValves;
+    }
+
+    public void setCountOfValves(int countOfValves) {
+        this.countOfValves = countOfValves;
+    }
+
+    public double getCompressionRatio() {
+        return compressionRatio;
+    }
+
+    public void setCompressionRatio(double compressionRatio) {
+        this.compressionRatio = compressionRatio;
+    }
+
+    public String getTypeOfFuel() {
+        return typeOfFuel;
+    }
+
+    public void setTypeOfFuel(String typeOfFuel) {
+        this.typeOfFuel = typeOfFuel;
+    }
+
+    @Override
+    public Engine clone() throws CloneNotSupportedException {
+        return new Engine(this.getTypeOfEngine(),
```

Comment:
try to use fields itself instead of getters

Reply:
Ok. I thought this solution was more right.